### ISSUE TITLE
Include full check name alongside ID in XML violation reports

### ISSUE THREAD

**Issue by Anmol202005** [checkstyle/checkstyle#18088]
From https://github.com/checkstyle/checkstyle-openrewrite-recipes/pull/123

## Description

When a Checkstyle module is configured with a custom `id`, the XML output shows **only the ID** in the `source` attribute. This makes it impossible to know which actual check raised the violation.

## Example Configuration

```xml
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">

    <!-- Instance WITHOUT ID -->
    <module name="NewlineAtEndOfFile"/>

    <!-- Instance WITH ID -->
    <module name="NewlineAtEndOfFile">
        <property name="id" value="twCheck"/>
    </module>

</module>
```

### Test.java

```bash
checkstyle/autofix/recipe ❯ cat Test.java                                                                                                                                                       java    06:34 
package org.checkstyle.autofix.recipe;

public class Test {
}
```

### Output
```bash

 checkstyle/autofix/recipe ❯javaa -jarcheckstyle-12.1.2-all.jarr \                                                                                                                               java    06:22 
  -c config.xml \
  -f xml \
  Test.java

<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="12.1.2">
<file name="/home/tomato/MINE/checkstyle-openrewrite-recipes/src/main/java/org/checkstyle/autofix/recipe/Test.java">
<error line="1" severity="error" message="File does not end with a newline." source="com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck"/>
<error line="1" severity="error" message="File does not end with a newline." source="twCheck"/>
</file>
</checkstyle>
Checkstyle ends with 2 errors.
```
### Problem

When id is used, the source attribute becomes ambiguous: `source="twCheck"` 
There is no way to know which check this refers to.
Tools consuming XML cannot map violations back to their actual modules.

SARIF already solves this problem by using: `source="com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck#twCheck"`

**Comment by romani** [comment]
Thanks a lot for detailed report.
Please add print of whole sarif report in issue description with CLI how to render it, to be 100% exact and transparent.

**Comment by romani** [comment]
@ivamly , please help us to fix this issue, it is blocker for our autofix project https://github.com/checkstyle/checkstyle-openrewrite-recipes

**Comment by ivamly** [comment]
I am on it.

**Comment by romani** [comment]
Fix is merged, will be released in a week
